tests/integration/questionnaire/test_questionnaire_list_collector_primary_person.py

```diff
@@ -131,3 +131,36 @@ def test_changing_answer_from_no_to_yes_on_primary_person_list_collector_resumes
         # Then on resuming, I am returned to the primary-person-list-collector
         self.launchSurvey("test_list_collector_primary_person", reponse_id=response_id)
         self.assertInUrl("/questionnaire/primary-person-list-collector/")
+
+    def test_adding_non_primary_person_using_second_list_collector_when_no_primary(
+        self,
+    ):
+        self.launchSurvey("test_list_collector_primary_and_two_list_collectors")
+
+        self.post({"you-live-here": "No"})
+
+        self.assertInBody("Does anyone else live at your address?")
+
+        self.post({"anyone-usually-live-at-answer": "No"})
+
+        self.assertInBody("Does anyone else live here?")
+
+        self.post({"anyone-else": "No"})
+
+        self.assertInBody("Does anyone else live at your address?")
+
+        self.post({"another-anyone-usually-live-at-answer": "Yes"})
+
+        self.assertInBody("What is the name of the person?")
+
+        self.post({"first-name": "Marie", "last-name": "Day"})
+
+        self.post({"another-anyone-else": "No"})
+
+        self.assertInUrl("/questionnaire/sections/section/")
+
+        first_person_change_link = self.get_link(1, "Change")
+
+        self.get(first_person_change_link)
+
+        self.assertInBody("Change details for Marie Day")
```

Comment:
This also needs the remove link tested

Reply:
Added remove test so it should be all covered now👍